src/scikit_build_core/settings/metadata.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import pathlib
+import warnings
+
+from pyproject_metadata import StandardMetadata
+
+from .._compat import importlib, tomllib
+from ..settings.skbuild_model import ScikitBuildSettings
+
+__all__ = ["setuptools_scm_version", "fancy_pypi_readme", "get_standard_metadata"]
+
+
+def __dir__() -> list[str]:
+    return __all__
+
+
+def setuptools_scm_version(pyproject_path: pathlib.Path) -> str:
+    from setuptools_scm import Configuration, _get_version
+
+    config = Configuration.from_file(str(pyproject_path))
+    version: str = _get_version(config)
+
+    return version
```

Comment:
@RonnyPfannschmidt Would you be okay with a public way to access `_get_version`? The problem is it's really had to respect `tool.setuptools_scm`'s config without it, and we don't want to have to reinvent new names for everything on every tool (like hatch-vcs does). It would be nice to just let people configure setuptools_scm the way it's normally configured by reading it's docs, and then we just ask it for the version.

Reply:
I almost have that ready on the current mainline, it can be integrated into the next release

Note that I'm also preparing to supersed setuptools_scm with vcs-versioning within the next couple of weeks

Reply:
Sweet, thanks! Will it avoid the file finder hook that makes setuptools_scm so hard to use for core packaging tools? Thinks like Spack aren't smart enough to build things in isolation, so a single package depending on setuptools_scm causes every build of every unrelated package to suddenly get different files.

For this PR, we can check the setuptools_scm version, and use the new version if present - it's safer to use private API if it's only for old versions. We also might want to wait on (at least advertising) this until vcs-versioning is out. There's an experimental flag we could hide it behind.

Reply:
Setuptools_scm currently has no easy / simple way to opt out of file finders

It will still trigger,

Maybe its time to add a contextvar to enable the opt in/out

Reply:
If `vcs-versioning` didn't have the setuptools plugin entrypoint, and only `setuptools_scm` did, that would already be very helpful. hatch-vcs and our usage could only depend on the non-invasive `vcs-versioning`. Not a complete fix (package that want to use the SCM discovery would still leak `setuptools_scm`), but it would help.

Reply:
the rough plan is to provide a file-finder extra and only enable the file finder if the current project has the extra in the build deps